newton/_src/solvers/kamino/_src/solvers/dvi/kernels.py

```
@@ -0,0 +1,1003 @@
+
+            # Equality constraints use scalar Gauss-Seidel updates. This keeps the
+            # solve on Kamino's Delassus system while avoiding a separate subsystem.
+            for i in range(njc):
+                v_i = _compute_row_velocity(ncts, mio, vio, i, problem_D, problem_v_f, solution_lambdas)
+                D_ii = wp.abs(problem_D[mio + ncts * i + i]) + cfg.regularization + FLOAT32_EPS
+                delta = -cfg.omega * v_i / D_ii
+                solution_lambdas[vio + i] += delta
+                max_step = wp.max(max_step, wp.abs(delta))
+                max_velocity = wp.max(max_velocity, wp.abs(v_i))
+
+            for l in range(nl):
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)